src/hotspot/share/opto/compile.hpp

```diff
@@ -371,6 +371,7 @@ class Compile : public Phase {
 
   // Compilation environment.
   Arena                 _comp_arena;            // Arena with lifetime equivalent to Compile
+  ResourceArea          _idealloop_arena;       // For data whose lifetime is a pass of loop optimizations
```

Comment:
```suggestion
  ResourceArea          _idealloop_arena;       // For data whose lifetime is a single pass of loop optimizations
``` 
Nit: This makes it abundantly clear that the data is freed after one pass.